src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -31,0 +1,75 @@
+import React, { useState } from 'react';
+
+type SelectComponentProps = {
+  handleSelectChange: (value: string) => void;
+  query: string;
+  handleQueryChange: (value: string) => void;
+  clearQuery: () => void;
+};
+
+export const TodoFilter: React.FC<SelectComponentProps> = ({
+  handleSelectChange,
+  query,
+  handleQueryChange,
+  clearQuery,
+}) => {
+  const [selectedValue, setSelectedValue] = useState<string>('all');
+
+  const handlChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = event.target.value;
+
+    setSelectedValue(value);
+    handleSelectChange(value);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    handleQueryChange(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedValue}
+            onChange={handlChange}
```

Comment:
The onChange handler is using a function with a typo in its name ('handlChange'). It should be corrected to 'handleChange'.